File: app/src/main/java/com/example/mycapstone/history/db/HistoryDao.java
package com.example.mycapstone.history.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface HistoryDao {

    @Insert
    void insert(History history);

    @Query("SELECT * FROM history ORDER BY id DESC LIMIT :limit OFFSET :offset")
    List<History> getPagedHistory(int limit, int offset);

    @Delete
    void deleteHistory(History history);
}

File: app/src/main/java/com/example/mycapstone/history/db/HistoryDao.kt
package com.example.mycapstone.history.db

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface HistoryDao {
    @Insert
    suspend fun insert(history:History)

    @Query("SELECT * FROM history ORDER BY id DESC")
    suspend fun getAllHistory(): List<History>

    @Delete
    suspend fun deleteHistory(history: History)
}